### DISCUSSION TITLE
Feature Request: Databricks Delta Lake Plug-in

### DISCUSSION THREAD

**Discussion by Matt-Mustard** [grafana/grafana#47187] [Data Sources]
This is a feature request for a Databricks Delta Lake (based on Apache Parquet) plug-in for Grafana. 

This plug-in should allow a Grafana Dashboard to query a Databricks "delta table" as it would a MySQL table. For background, Databricks creates/updates/deletes the delta table, but the storage takes place on Azure/AWS/GCP storage offerings. In other solutions like the Power BI one below, client services can query the storage account through the Databricks cluster/compute resource (that is my understanding anyway).

### Current Options

**Power BI with Databricks Partner Connect**

Such an integration already exists for Power BI, but Power BI is not vendor agnostic and requires the costly premium licence for high refresh rates (useful for IoT and other streaming applications):

https://docs.microsoft.com/en-us/azure/databricks/integrations/bi/power-bi

**Alternative Azure solution:**

Currently if you wish to expose data that was processed by Databricks to Grafana, you would have to go via the Azure Data Explorer which introduces another vendor specific service.

https://docs.microsoft.com/en-us/azure/data-explorer/grafana

### Conclusion

As the compute and storage are already provided by Databricks and Azure Datalake/AWS S3, the proposed plug-in will authenticate to and leverage these existing resources without the need for providing a compatible data source such as Azure Data Factory, InfluxDB, etc.

I understand this may require a collaboration between the Grafana, Databricks and Azure/AWS/GCP teams, but I believe it would provide a large user base with an elegant solution for visualising their data in a streamlined fashion.

Thank you for taking the time to read this feature request. If a solution already addresses this, please let me know!

A similar request/question has been raised in the following locations by other users, but as this is now the best place for feature requests I thought I would create a new one:

https://community.grafana.com/t/create-grafana-dashboard-using-for-parquet-files-are-datasource/24441
https://stackoverflow.com/questions/56884767/need-solution-to-integrate-grafana-with-azure-data-lake

**Other References:**

https://databricks.com/
https://delta.io/
https://parquet.apache.org/

**Comment by Matt-Mustard** [comment]
### Databricks Partner Connect Program

This program appears to be the recommended route for software vendors to develop integrations with Databricks. Hope this helps.

https://databricks.com/partnerconnect

**Comment by epa095** [comment]
There now exists a grafana enterprise plugin for databricks SQL. It connects to a databricks sql endpoint (so a bit more costly than querying the delta table directly), but since it is an enterprise plugin it is a bit costly (and not open source). Since there now exists databricks SQL connectors for both Golang and node it should be quite easy to write a open source version. https://grafana.com/grafana/plugins/grafana-databricks-datasource/

**Comment by epa095** [comment]
Now there is also an open source databricks datasource https://github.com/mullerpeter/databricks-grafana

**Comment by mellieA** [comment]
Closing as resolved because we have an enterprise plugin for databricks and it looks like that someone created an open-source version of it as well.
